[Mrinal Ganesh] iP

mrmrinal opened the pull request:
## JarvisPro

> “Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes/)

JarvisPro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- <strike>FAST</strike> _SUPER_ FAST to use

All you need to do is,

1. download it 
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

mrmrinal committed:
Level - 1

mrmrinal committed:
Level-2

mrmrinal committed:
Added Task class and done method

mrmrinal committed:
Added the three classes, and Level 4 changes

mrmrinal committed:
Added Error handling and tried running with test cases

mrmrinal committed:
Added delete functionality, and updated input and expected

mrmrinal committed:
Made List item final

mrmrinal committed:
Commiting to add a UI Testing tag

mrmrinal committed:
Added Enums to the program and made seperate methods for recurring method calls.

mrmrinal committed:
Enable hard disk save

mrmrinal committed:
Added date recognition

mrmrinal committed:
Merge branch 'branch-Level-7' to 'master'

mrmrinal committed:
Merge branch 'branch-Level-8' to 'master'

mrmrinal committed:
Add ui class

mrmrinal committed:
Add Storage and TaskList

mrmrinal committed:
Modify Tasklist Remove Items

mrmrinal committed:
Add Parser

mrmrinal committed:
Add Package

mrmrinal committed:
Add JUnit Support

mrmrinal committed:
Add Javadocs

mrmrinal committed:
Add Coding Standards

mrmrinal committed:
Add Find functionality

mrmrinal committed:
Merge branch 'branch-A-JavaDoc'

mrmrinal committed:
Merge branch 'branch-A-CodingStandard'

mrmrinal committed:
Merge branch 'branch-Level-9'

mrmrinal committed:
Fix find method

mrmrinal committed:
Merge remote-tracking branch 'origin/add-gradle-support'

mrmrinal committed:
Add checkstyle

mrmrinal committed:
Add suppressions.xml checkstyle file

mrmrinal committed:
Add GUI

didymusne commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,48 @@
+import javafx.scene.control.Label;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+
```
Should there be a descriptive header for the public class?

didymusne commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,48 @@
+        this.getChildren().setAll(tmp);
+    }
+
+    public static DialogBox getUserDialog(Label l, ImageView iv) {
```
Should there be a descriptive header for the public method getUserDialog? I noticed this in several other places as well...

didymusne commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,361 @@
+ */
+public class Duke extends Application{
+
+    private final Storage storage;
```
I like how this is private and final!

didymusne commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,361 @@
+    private Image userimg = new Image(this.getClass().getResourceAsStream("/images/user.png"));
+    private Image dukeimg = new Image(this.getClass().getResourceAsStream("/images/nestor.png"));
+
+     public enum RequestType {
```
Should the spaces here be 4 instead of 5?

didymusne commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,361 @@
+
+    }
+    
+    private String done(String userInput) {
```
Should the method name be a verb? I noticed this at many other places as well...

zacharylwy replied:
This should be named a verb! but no big issue

didymusne commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,361 @@
+        }
+    }
+    
+    private String todo(String userInput) {
```
Should the method name be a verb? 

Perhaps, can consider using ENUM for the Tasks todo, event, deadline? (since quite a number of code appears repeated)

zacharylwy replied:
I guess depending on how one interprets this, this may seem to be named in an undesired manner (should be verb or action).

didymusne commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,40 @@
+/**
+ * A parser which aids with deciphering the nature of the user input.
+ *
+ * @author mrmrinal
```
I like the @author indication!

didymusne commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,44 @@
+    /**
+     * Bot greeting which shows up whenever the bot is instantiated.
+     */
+    public void dukeGreeting() {
```
Should the method name be a verb?

didymusne commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,7 @@
+package duke.exception;
+
+public class DukeException extends Exception{
+  public DukeException(String errorMessage){
```
Should there be 4 spaces instead of 2?

didymusne commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+     * @param done Status of whether task was done or not
+     */
+    public Deadline(String description, String by, int done) {
+      super(description, done);
```
Should there be 8 white spaces instead of 6?

didymusne reviewed (approved):
LGTM! Apart from some coding standard violations, it looks good and I like the tidiness of the code in general! Good job!

mrmrinal committed:
Add xml and style decoration

mrmrinal committed:
Merge branch 'branch-Level-10'

mrmrinal committed:
Remove message addition after 'bye' command

zacharylwy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,128 @@
+/**
+ * A class that helps in storing the class offline.
+ *
+ * @author mrmrinal
```
Nice @author tag

zacharylwy commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,7 @@
+package duke.exception;
+
+public class DukeException extends Exception{
```
{ should be preceded with whitespace.

zacharylwy reviewed (commented):
Looks great to me! keep up the good work

mrmrinal committed:
Add assert statements

mrmrinal committed:
Add Coding Quality

mrmrinal committed:
Merge pull request #1 from mrmrinal/branch-A-Assertions

Add assert statements

mrmrinal committed:
Merge branch 'master' into branch-A-CodeQuality

mrmrinal committed:
Merge pull request #2 from mrmrinal/branch-A-CodeQuality

Add Coding Quality

mrmrinal committed:
Add Update extension

mrmrinal committed:
Merge branch 'branch-BCD-Extension'

mrmrinal committed:
Add CI workflow

mrmrinal committed:
Edit CI workflow file

mrmrinal committed:
Fix file permissions for gradlew

mrmrinal committed:
Update OS for test

mrmrinal committed:
Update Method Names

Method names were not consistent with CodingStandards and had to be
renamed.

Changing the method names is more code friendly as now people who read
this code have a better understanding of what the methods do.

Let's
* Update method names were renamed to more explicitly convey its
  function

mrmrinal committed:
Fix Duke.java checkstyle

mrmrinal committed:
Fix checkstyle errors

mrmrinal committed:
Fix code quality for all classes

mrmrinal committed:
Add Ui.png

mrmrinal committed:
Update README.md

mrmrinal committed:
Set theme jekyll-theme-cayman

mrmrinal committed:
Set theme jekyll-theme-merlot

mrmrinal committed:
Set theme jekyll-theme-cayman

mrmrinal committed:
Fix javafx java version

mrmrinal committed:
Update _config.yml

mrmrinal committed:
Set theme jekyll-theme-tactile
